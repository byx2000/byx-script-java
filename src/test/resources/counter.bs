// 简单计数器
function SimpleCounter() {
    var cnt = 0
    return () => {
        cnt = cnt + 1
        return cnt
    }
}

var c1 = SimpleCounter()
Console.println(c1())
Console.println(c1())
Console.println(c1())

var c2 = SimpleCounter()
Console.println(c2())
Console.println(c2())
Console.println(c2())

Console.println(c1())
Console.println(c2())

// 计数器
function Counter(init) {
    var cnt = init
    return {
        // 获取当前计数值
        current: () => cnt,
        // 计数值+1
        inc: () => {cnt++},
        // 计数值-1
        dec: () => {cnt--}
    }
}

var c3 = Counter(100)
Console.println(c3.current()) // 100
c3.inc()
Console.println(c3.current()) // 101
c3.inc()
Console.println(c3.current()) // 102
c3.dec()
Console.println(c3.current()) // 101
c3.dec()
Console.println(c3.current()) // 100

var c4 = Counter(200)
Console.println(c4.current()) // 200
c4.inc()
Console.println(c4.current()) // 201
c4.inc()
Console.println(c4.current()) // 202
c4.dec()
Console.println(c4.current()) // 201
c4.dec()
Console.println(c4.current()) // 200
