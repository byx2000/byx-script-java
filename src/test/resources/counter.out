1
2
3
1
2
3
4
4
100
101
102
101
100
200
201
202
201
200
